components/script_bindings/codegen/codegen.py

```diff
@@ -2275,30 +2283,44 @@ class CGImports(CGWrapper):
     """
     Generates the appropriate import/use statements.
     """
-    def __init__(self, child, descriptors, callbacks, dictionaries, enums, typedefs, imports, config):
+    def __init__(
+            self,
+            child: CGThing,
+            descriptors: list[Descriptor],
+            callbacks: list[IDLCallback],
+            dictionaries: list[IDLDictionary],
+            enums: list[IDLEnum],
+            typedefs: list[IDLTypedef],
+            imports: list[str],
+            config: Configuration,
+    ) -> None:
         """
         Adds a set of imports.
         """
 
-        def componentTypes(type):
+        def componentTypes(type: IDLType) -> list[IDLType]:
             if type.isType() and type.nullable():
                 type = type.unroll()
             if type.isUnion():
+                assert isinstance(type, IDLUnionType)
+                assert type.flatMemberTypes is not None
                 return type.flatMemberTypes
             if type.isDictionary():
-                return [type] + getTypesFromDictionary(type)
+                assert isinstance(type, (IDLDictionary, IDLWrapperType))
+                return [cast(IDLType, type)] + getTypesFromDictionary(type)
             if type.isSequence():
+                assert isinstance(type, IDLSequenceType)
                 return componentTypes(type.inner)
             return [type]
 
-        def isImportable(type):
+        def isImportable(type: IDLType) -> bool:
             if not type.isType():
                 assert (type.isInterface() or type.isDictionary()
                         or type.isEnum() or type.isNamespace())
                 return True
             return not (type.builtin or type.isSequence() or type.isUnion())
 
-        def relatedTypesForSignatures(method):
+        def relatedTypesForSignatures(method: IDLMethod) -> list[IDLType]:
```

Comment:
```suggestion
        def relatedTypesForSignatures(method: IDLMethod | IDLCallback) -> list[IDLType]:
```

This should allow removing the cast [here](https://github.com/servo/servo/blob/3dc91841211285afa2840c871e69d43ac89b9a55/components/script_bindings/codegen/codegen.py#L2359)